index.html

```diff
@@ -4820,6 +4820,113 @@ <h3>Verification</h3>
 
       </section>
 
+      <section class="normative">
+        <h3>Context Validation</h3>
+
+        <p>
+This section contains an algorithm that [=verifiers=] MUST run after running the
+algorithm in Section [[[#verification]]], when validating a [=verifiable
+credential=] or a [=verifiable presentation=]. This algorithm takes a document
+([=map=] |inputDocument|) and a set of approved JSON-LD Contexts ([=map=]
+|approvedContexts|) and returns a [=map=] that contains the following:
+        </p>
+
+        <ul>
+          <li>
+a status ([=boolean=] |status|)
+          </li>
+          <li>
+a [=conforming document=] ([=map=] |document|)
+          </li>
+          <li>
+zero or more warnings ([=list=] of [=ProblemDetails=] |warnings|)
+          </li>
+          <li>
+zero or more errors ([=list=] of [=ProblemDetails=] |errors|)
+          </li>
+        </ul>
+
+        <p>
+The context validation algorithm is as follows:
+        </p>
+
+        <ol class="algorithm">
+          <li>
+Set |result|.|status| to `false`.
+          </li>
+          <li>
+For each |contextValue| expressed in every `@context` property, ensure that
+either:
+            <ol class="algorithm">
+              <li>
+The |contextValue| exists in |approvedContexts| and matches a cryptographic
+hash associated with the matching approved context, or
+              </li>
+              <li>
+optionally, for the context values that do not exist in |approvedContexts|,
+remove the |contextValue| from the `@context` property values. Set
+|result|.|document| to the result of running the
+<a data-cite="JSON-LD11-API#compaction-algorithm">
+JSON-LD Compaction Algorithm</a> with the modified |inputDocument| as input.
```

Comment:
Hmm, this isn't quite right.

```suggestion
optionally, if any of the context values do not exist in |approvedContexts|,
clone the `@context` property and remove each context value that was
not in |approvedContext| from the clone. Set |result|.|document| to
the result of running the <a data-cite="JSON-LD11-API#compaction-algorithm">
JSON-LD Compaction Algorithm</a> with the |inputDocument| and
the cloned and modified context as input.
```